flood/src/main/java/com/dsvl/flood/service/SearchService.java

```diff
@@ -0,0 +1,12 @@
+package com.dsvl.flood.service;
+
+import com.dsvl.flood.File;
+import com.dsvl.flood.Neighbour;
+
+import java.net.InetAddress;
+import java.util.List;
+
+public interface SearchService {
+
+    List<File> search(String fileName, List<Neighbour> neighbours, InetAddress nodeAddress, Integer nodePort);
```

Comment:
Since the `node` has two ports (`UDP` and `TCP`), it is better to refactor `nodePort` as `nodeTcpPort`